tools/odbc/duckdb_odbc.cpp

```diff
@@ -1,15 +1,26 @@
 #include "duckdb_odbc.hpp"
 #include "odbc_fetch.hpp"
 
-// duckdb::OdbcHandle::~OdbcHandle() = default;
+using duckdb::OdbcHandleStmt;
 
-duckdb::OdbcHandleStmt::OdbcHandleStmt(OdbcHandleDbc *dbc_p)
+OdbcHandleStmt::OdbcHandleStmt(OdbcHandleDbc *dbc_p)
     : OdbcHandle(OdbcHandleType::STMT), dbc(dbc_p), rows_fetched_ptr(nullptr) {
 	D_ASSERT(dbc_p);
 	D_ASSERT(dbc_p->conn);
 
 	odbc_fetcher = make_unique<OdbcFetch>();
+	dbc->stmt_handle = this;
```

Comment:
Is there a situation in which the `stmt_handle` would be deleted/destroyed? e.g. what would happen if the user would call `SQLFreeHandle` on the ODBC statement, and would afterwards call `SQLEndTran(SQL_COMMIT)`? Would `dbc->stmt_handle` point to a `OdbcHandleStmt` that has been deleted in this case?

Reply:
At SQLEndTran line 111 there is a verification to check if the dbc->stmt_handle is valid, that is also used to check when there is no associated stmt_handle to the OdbcHandleDbc.

In addition, I think is needed to set to null the reference of the released stmt_handle when calling SQLFreeHandle, e.g.:

```
	case SQL_HANDLE_STMT: {
		auto *hdl = (duckdb::OdbcHandleStmt *)handle;
                hdl->dbc->stmt_handle = nullptr;
		delete hdl;
        }
```

I think that is enough to avoid such problem, see https://github.com/duckdb/duckdb/pull/2109/commits/90049fc32d4f07e95c0254ec77ac9aca186a2063.